Dodge.js

```diff
@@ -0,0 +1,186 @@
+import { StatusBar } from 'expo-status-bar';
+import { StyleSheet, Text, View, Dimensions, Button } from 'react-native';
+import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import { GameEngine } from "react-native-game-engine";
+import AxisPad from 'react-native-axis-pad';
+
+//https://reactnative.dev/docs/timers.html
+//this.setTimeout??
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+let index = 0
```

Comment:
This isn't a great approach to unique identifiers. Having one global variable means things go very wrong if you ever update it incorrectly (and in fact with timers, and multiple functions, correct updates could just happen in the wrong order and confuse everything).

I suggest assigning a unique identifier to each entity when it is created (e.g. https://dev.to/rahmanfadhil/how-to-generate-unique-id-in-javascript-1b13). It might also make sense to store the id somewhere inside the entity itself.

If you have trouble debugging entities later, you can even leave them in the entities object and just "deactivate" them (have a variable like "alive = true").